lib/model/message.dart

```diff
@@ -125,10 +125,21 @@ class MessageStoreImpl with MessageStore {
     if (message == null) return;
 
     message.flags = event.flags;
+    // According to the API, the content of the message is edited if origContent
+    // is not null.
+    if (event.origContent != null) {
+      message.editState = MessageEditState.edited;
+    }
     if (event.renderedContent != null) {
       assert(message.contentType == 'text/html',
         "Message contentType was ${message.contentType}; expected text/html.");
       message.content = event.renderedContent!;
+      if (
+        !(event.renderingOnly == true
+        || event.userId == null) // TODO(server-5) can remove userId check
+      ) {
+        message.editState = MessageEditState.edited;
```

Comment:
Hmm I see.

How does this interact with the `origContent` check above? Do we need both checks?

In light of this detail about `renderingOnly`, is it correct to have that `origContent` logic? It seems like it might cause us to record the state as "edited" when in fact it was a rendering-only change.

Reply:
More generally: I think it's probably best for the updates to `editState` to live in one place. In other words, to organize the updates in this function by what they're updating, rather than by what data they're consulting. I think that gives the easiest setup for thinking through each update's logic to check whether it's right.

Reply:
Checking the server code, `origContent` only appears when there is an actual update. So I think it is safe to assume that the second is not needed.

Reply:
Hmm interesting. That's not how I would have interpreted the docs:
> orig_content: string
> Only present if this event changed the message content.

But I guess it can be read the other way too, using "changed the message content" to mean that it wasn't a rendering-only update either.

… Oh, and comparing with what it says at `content`:
> Only present if this event changed the message content or updated the message content for an [inline URL preview](https://zulip.readthedocs.io/en/latest/subsystems/sending-messages.html#inline-url-previews).

I guess the docs are making a contrast there.

So in short I agree this interpretation seems sound. I'll start a thread in `#api documentation` to see about clarifying the docs.